File: backend/rorapp/models/senator_action_log.py
from django.db import models
from rorapp.models.senator import Senator
from rorapp.models.action_log import ActionLog


# Model for representing relationships between senators and action_logs

class SenatorActionLog(models.Model):
    senator = models.ForeignKey(Senator, on_delete=models.CASCADE)
    action_log = models.ForeignKey(ActionLog, on_delete=models.CASCADE)

File: frontend/classes/SenatorActionLog.ts
interface SenatorActionLogData {
  id: number
  senator: number
  action_log: number
}

class SenatorActionLog {
  id: number
  senator: number
  action_log: number

  constructor(data: SenatorActionLogData) {
    this.id = data.id
    this.senator = data.senator
    this.action_log = data.action_log
  }
}

export default SenatorActionLog
